RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,462 @@
+    desc.add<int>("maxNVertices", 1);
+
+    desc.add<double>("ptMin", 0.9);
+    desc.add<double>("originRadius", 0.2);
+    desc.add<double>("zErrorBeamSpot", 24.2);
+    desc.add<double>("deltaEta", 0.5);
+    desc.add<double>("deltaPhi", 0.5);
+    desc.add<double>("deltaEta_Cand", -1.);
+    desc.add<double>("deltaPhi_Cand", -1.);
+    desc.add<double>("deltaEta_Point", -1.);
+    desc.add<double>("deltaPhi_Point", -1.);
+    desc.add<bool>("precise", true);
+
+    desc.add<double>("nSigmaZVertex", 3.);
+    desc.add<double>("zErrorVetex", 0.2);
+    desc.add<double>("nSigmaZBeamSpot", 4.);
+
+    desc.add<std::string>("whereToUseMeasurementTracker", "ForSiStrips");
+    desc.add<edm::InputTag>("measurementTrackerName", edm::InputTag(""));
+
+    desc.add<bool>("searchOpt", false);
+
+    // Only for backwards-compatibility
+    edm::ParameterSetDescription descRegion;
+    descRegion.add<edm::ParameterSetDescription>("RegionPSet", desc);
+
+    descriptions.add("candidatePointSeededTrackingRegionsFromBeamSpot", descRegion);
+  }
+
+
+  std::vector<std::unique_ptr<TrackingRegion> > regions(const edm::Event& e, const edm::EventSetup& es) const override
+  {
+    std::vector<std::unique_ptr<TrackingRegion> > result;
+
+    // pick up the candidate objects of interest    
+    edm::Handle< reco::CandidateView > objects;
+    size_t n_objects = 0;
+
+    if(m_seedingMode == SeedingMode::CANDIDATE_SEEDED || m_seedingMode == SeedingMode::CANDIDATE_POINT_SEEDED){
+      e.getByToken( token_input, objects );
+      n_objects = objects->size();
+      if (n_objects == 0) return result;
+    }
+
+    const auto& objs = *objects;
+
+    // always need the beam spot (as a fall back strategy for vertex modes)
+    edm::Handle< reco::BeamSpot > bs;
+    e.getByToken( token_beamSpot, bs );
+    if( !bs.isValid() ) return result;
+
+    // this is a default origin for all modes
+    GlobalPoint default_origin( bs->x0(), bs->y0(), bs->z0() );
+
+    // vector of origin & halfLength pairs:
+    std::vector< std::pair< GlobalPoint, float > > origins;
+
+    // fill the origins and halfLengths depending on the mode
+    if (m_mode == Mode::BEAM_SPOT_FIXED || m_mode == Mode::BEAM_SPOT_SIGMA)
+    {
+      origins.emplace_back( default_origin,
+			    (m_mode == Mode::BEAM_SPOT_FIXED) ? m_zErrorBeamSpot : m_nSigmaZBeamSpot*bs->sigmaZ()
+			    );
+    }
+    else if (m_mode == Mode::VERTICES_FIXED || m_mode == Mode::VERTICES_SIGMA)
+    {
+      edm::Handle< reco::VertexCollection > vertices;
+      e.getByToken( token_vertex, vertices );
+      int n_vert = 0;
+      for ( const auto& v : (*vertices) )  
+      {
+        if ( v.isFake() || !v.isValid() ) continue;
+	origins.emplace_back( GlobalPoint( v.x(), v.y(), v.z() ),
+			      (m_mode == Mode::VERTICES_FIXED) ? m_zErrorVetex : m_nSigmaZVertex*v.zError()
+			      );
+        ++n_vert;
+	if( !(n_vert < m_maxNVertices) ) break;
```

Comment:
Could this and the other occurrances of the same pattern be changed to `if(n_vert >= m_maxNVertices) break;`? Avoiding `!` makes it easier to read (to me at least).

Reply:
I agree